File: Solution in CPP/Que10.cpp
// Check if a number is prime 
#include<iostream>
using namespace std;

bool isPrime(int number) {
    if(number <= 1) {
        return false; 
    }
    
    for(int i = 2; i*i<=number; ++i) {
        if(number%i == 0) {
            return false;
        }
    }


    return true;
}

int main() {
    int number = 29;

    if(isPrime(number)) {
        cout<<number<<" is a prime number"<<endl;
    } 
    else {
        cout<<number<<" is not a prime number"<<endl;
    }
    return 0;
}

File: Solution in RUST/Que10.rs
// Check if a number is prime in Rust


fn is_prime(number: u32) -> bool {
    if number <= 1 {
        return false;
    }

    let sqrt_num = (number as f64).sqrt() as u32;
    for i in 2..=sqrt_num {
        if number % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    let number = 29;

    if is_prime(number) {
        println!("{} is a prime number", number);
    } else {
        println!("{} is not a prime number", number);
    }
}
